src/main/java/com/minecolonies/entity/ai/citizen/guard/EntityAIMeleeGuard.java

```diff
@@ -0,0 +1,211 @@
+          new AITarget(GUARD_SEARCH_TARGET, this::searchTarget),
+          new AITarget(GUARD_GET_TARGET, this::getTarget),
+          new AITarget(GUARD_HUNT_DOWN_TARGET, this::huntDown),
+          new AITarget(GUARD_PATROL, this::patrol),
+          new AITarget(GUARD_RESTOCK, this::goToBuilding)
+        );
+
+        if (worker.getCitizenData() != null)
+        {
+            worker.setSkillModifier(2 * worker.getCitizenData().getIntelligence() + worker.getCitizenData().getStrength());
+            worker.setCanPickUpLoot(true);
+        }
+    }
+
+    @Override
+    protected AIState searchTarget()
+    {
+        if (checkForWeapon())
+        {
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        InventoryFunctions.matchFirstInInventory(worker.getInventoryCitizen(), stack -> stack != null && Utils.doesItemServeAsWeapon(stack), worker::setHeldItem);
+        return super.searchTarget();
+    }
+
+    /**
+     * Follow the target and kill it.
+     *
+     * @return the next AIState.
+     */
+    protected AIState huntDown()
+    {
+        if (!targetEntity.isEntityAlive() || checkForWeapon())
+        {
+            targetEntity = null;
+        }
+
+        if (targetEntity != null)
+        {
+            if (worker.getEntitySenses().canSee(targetEntity) && worker.getDistanceToEntity(targetEntity) <= MIN_ATTACK_DISTANCE)
+            {
+                worker.resetActiveHand();
+                attackEntity(targetEntity, (float) DAMAGE_PER_ATTACK);
+                setDelay(getReloadTime());
+                attacksExecuted += 1;
+
+                if (attacksExecuted >= getMaxAttacksUntilRestock())
+                {
+                    return AIState.GUARD_RESTOCK;
+                }
+
+                return AIState.GUARD_HUNT_DOWN_TARGET;
+            }
+            worker.setAIMoveSpeed((float) (BASE_FOLLOW_SPEED + BASE_FOLLOW_SPEED_MULTIPLIER * worker.getExperienceLevel()));
+            worker.isWorkerAtSiteWithMove(targetEntity.getPosition(), (int) MIN_ATTACK_DISTANCE);
+
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+
+        worker.setAIMoveSpeed(1);
+        return AIState.GUARD_SEARCH_TARGET;
+    }
+
+    private void attackEntity(@NotNull EntityLivingBase entityToAttack, float baseDamage)
+    {
+        double damgeToBeDealt = baseDamage;
+
+        ItemStack heldItem = worker.getHeldItem(EnumHand.MAIN_HAND);
+        if (heldItem != null)
+        {
+            if (heldItem.getItem() instanceof ItemSword)
+            {
+                damgeToBeDealt += ((ItemSword) heldItem.getItem()).getDamageVsEntity();
+            }
+            damgeToBeDealt += EnchantmentHelper.getModifierForCreature(heldItem, targetEntity.getCreatureAttribute());
+        }
+
+        targetEntity.attackEntityFrom(new DamageSource(worker.getName()), (float) damgeToBeDealt);
+        targetEntity.setRevengeTarget(worker);
+
+        int fireAspectModifier = EnchantmentHelper.getFireAspectModifier(worker);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'fireAspectModifier' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)